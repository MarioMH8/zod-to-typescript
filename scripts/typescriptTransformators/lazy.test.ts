import { ZodTypescriptTransformator } from "@scripts/index";
import { z as zod, type ZodType } from "zod";

it("lazy", () => {
	const zodSchema: ZodType = zod.object({
		prop1: zod.lazy(() => zodSchema.array()),
	});

	const result = ZodTypescriptTransformator.convert(zodSchema);

	const expected = "type Zod2ts_0_duplojs = { prop1: Zod2ts_0_duplojs[]; };".replace(/\s+/g, "");
	const received = result.replace(/\s+/g, "");
	expect(received).toBe(expected);
});
