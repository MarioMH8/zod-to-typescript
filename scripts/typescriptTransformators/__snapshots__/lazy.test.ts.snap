// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`lazy 1`] = `
"type Zod2ts_0_duplojs = {
    prop1: Zod2ts_0_duplojs[];
};"
`;
